plugins/inventory/iocage.py

```diff
@@ -0,0 +1,251 @@
+  release: iocage_release | split('-') | first
+groups:
+  test: inventory_hostname.startswith('test')
+keyed_groups:
+  - prefix: distro
+    key: iocage_release
+  - prefix: state
+    key: iocage_state
+'''
+
+import re
+from subprocess import Popen, PIPE
+
+from ansible.errors import AnsibleParserError
+from ansible.module_utils.common.text.converters import to_native, to_text
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+from ansible.utils.display import Display
+
+display = Display()
+
+
+def _parse_ip4(ip4):
+    if ip4 == '-':
+        return ip4
+    return re.split('\\||/', ip4)[1]
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+    ''' Host inventory parser for ansible using iocage as source. '''
+
+    NAME = 'community.general.iocage'
+    IOCAGE = '/usr/local/bin/iocage'
+
+    def __init__(self):
+        super(InventoryModule, self).__init__()
+
+    def verify_file(self, path):
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(('iocage.yaml', 'iocage.yml')):
+                valid = True
+            else:
+                self.display.vvv('Skipping due to inventory source not ending in "iocage.yaml" nor "iocage.yml"')
+        return valid
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(inventory, loader, path)
+
+        self._read_config_data(path)
+        cache_key = self.get_cache_key(path)
+
+        user_cache_setting = self.get_option('cache')
+        attempt_to_read_cache = user_cache_setting and cache
+        cache_needs_update = user_cache_setting and not cache
+
+        if attempt_to_read_cache:
+            try:
+                results = self._cache[cache_key]
+            except KeyError:
+                cache_needs_update = True
+        if not attempt_to_read_cache or cache_needs_update:
+            results = self.get_inventory(path)
+        if cache_needs_update:
+            self._cache[cache_key] = results
+
+        self.populate(results)
+
+    def get_inventory(self, path):
+        host = self.get_option('host')
+        env = self.get_option('env')
+        get_properties = self.get_option('get_properties')
+
+        cmd = []
+        if host != 'localhost':
+            user = self.get_option('user')
+            cmd.append(f"ssh {user}@{host}")
+        if env:
+            cmd.append(' '.join(env))
+        cmd.append(self.IOCAGE)
+
+        cmd_list = cmd.copy()
+        cmd_list.append('list --header --long')
+        cmd_list = ' '.join(cmd_list)
+        try:
+            p = Popen(cmd_list, shell=True, text=True, stdout=PIPE, stderr=PIPE)
+            stdout, stderr = p.communicate()
+            if p.returncode != 0:
+                raise AnsibleParserError('Failed to run cmd=%s, rc=%s, stderr=%s' %
+                                         (cmd_list, p.returncode, to_native(stderr)))
```

Comment:
Arguably that it not a parser error. That looks more like `AnsibleError` than `...ParserError`

Reply:
Why do `nmap.py` and `virtualbox.py` use `AnsibleParserError` to handle `Popen` errors?

Reply:
Most likely because I wasn't there ;-) Or I wasn't paying attention :-) Anyways, regardless of that, would you agree that this is not a parsing issue?

Reply:
You're right. It's not a parsing issue. I fixed the calls.